### PR TITLE
Add 'id' as a unique identifier on config-repo tag

### PR DESCRIPTION
**Pull request by GaneshSPatil**
* New config-repo tag example:

```
<config-repo plugin="json.config.plugin" id="9271a1c8-b1ee-4a76-868a-36ffaa349ecc">
	<git url="https://github.com/bob/gocd-json-config-example.git" />
</config-repo>
```

### CODE REVIEW

File: config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java [hunk 1]
```diff
@@ -71,6 +79,16 @@ public String getConfigProviderPluginName() {
         return configProviderPluginName;
     }
 
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        if(StringUtil.isBlank(id))
+            id = null;
```
Comment (varshavaradarajan) [inline config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java]: this should be randomly generated, no? See, ensureIdExists method in SCM.java

File: server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java [hunk 2]
```diff
@@ -1502,6 +1502,20 @@ public void shouldMigrateLdapToAuthConfigAsPartOfMigration91() throws Exception
     }
 
     @Test
+    public void shouldAddIdOnConfigRepoAsPartOfMigration94() throws Exception {
+        String configXml = "<cruise schemaVersion='93'>" +
+                "<config-repos>\n" +
+                "   <config-repo plugin=\"json.config.plugin\">\n" +
```
Comment (varshavaradarajan) [inline server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java]: Did you test with multiple config-repos?

**Comment by varshavaradarajan** [comment]
Can you also send a PR to functional tests to update the schema version? We can merge the two together.

File: server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java (hunk 2 above)
Comment (GaneshSPatil) [inline server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java]: @varshavaradarajan -- do we need to test it with multiple `config-repo` tags?
if yes, how it is different from testing with just one?

File: config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java (hunk 1 above)
Comment (ketan) [inline config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java]: We don't do that with elastic agent and auth plugins. Users are expected to provide an identifier.

**Comment by GaneshSPatil** [comment]
@varshavaradarajan -- We also need to remove the duplicate config-repo which is currently allowed.
We'll do it as part of the same PR

Do not merge until then

File: server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java (hunk 2 above)
Comment (varshavaradarajan) [inline server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java]: Its not. I am not saying write a test. Just asking you to confirm that there's no unexpected behavior when you have multiple config repos

File: config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java (hunk 1 above)
Comment (varshavaradarajan) [inline config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java]: Not sure if the xsd validation takes cate of things like id="                ". There should be a validation on the model for the id then.

**Pull request merged**